### PR TITLE
Add Araf

### PR DESCRIPTION
**Pull request by montelucci**

### CODE REVIEW

File: src/variety/araf.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+	decodeData : function(){
+		this.decodeCellQnum();
+		this.decodeBorderAns();
+	},
+	encodeData : function(){
+		this.encodeCellQnum();
+		this.encodeBorderAns();
+	},
+
+	kanpenOpen : function(){
```
Comment (robx) [inline src/variety/araf.js]: I don't think the kanpen-related code should be there (unless this is intentional and kanpen supports araf)? As far as I understand, kanpen is a different puzzle format for some of the more common types.

So all of `FileIO` except for `{de,en}codeData` can go, as well as `Encode:{de,en}codeKanpen`.

File: src/variety/araf.js [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+	bordercolor_func : "qans",
+
+	circleratio : [0.45, 0.40],
+
+
+	paint : function(){
+		this.drawBGCells();
+		this.drawDashedGrid();
+
+		// this.drawQuesNumbers();
```
Comment (robx) [inline src/variety/araf.js]: (this line could also go)

**Comment by robx** [comment]
Oh, could you also add a test case for `checkBorderDeadend`?

File: src/variety/araf.js [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+		this.encodeNumber16();
+	}
+},
+
+//---------------------------------------------------------
+// 正解判定処理実行部
+AnsCheck:{
+	checklist : [
+		"checkLessThan2Num",
+		"checkMoreThan2Num",
```
Comment (x-sheep) [inline src/variety/araf.js]: This is just my personal preference, but could you move this check below the `checkBigArea` and `checkSmallArea` lines? IMHO the checks for solutions which are (always) invalid should come before the checks for solutions which are unfinished.

Comment (montelucci) [reply src/variety/araf.js]: Fair

Comment (robx) [reply src/variety/araf.js]: While I agree with the general idea, I'm not sure it makes sense here? You can't check area size if there's not exactly two clues in an area. (As I'm writing this, test are failing with the reordered checklist, and I'm not sure how that could be fixed.)

Comment (robx) [reply src/variety/araf.js]: I suppose the too large/too small checks could check the areas with two clues only... But is the extra complexity worth it?

Comment (x-sheep) [reply src/variety/araf.js]: The checks already ignore areas without exactly two clues.

**Pull request merged**